autoemulate/experimental/emulators/base.py

```diff
@@ -168,3 +168,29 @@ def get_tune_config():
             "each subclass."
         )
         raise NotImplementedError(msg)
+
+
+class SklearnBackend(Emulator):
+    """
+    SklearnBackend is a sklearn model and implements the base class.
+    This provides default implementations to further subclasses.
+    This means that models can subclass and only need to implement
+    `.fit()` and `.predict()` to have an emulator to be run in `AutoEmulate`
+    """
+
+    def __init__(
+        self, x: InputLike | None = None, y: InputLike | None = None, **kwargs
+    ):
+        pass
+
+    def fit(self, x: InputLike, y: InputLike | None):
+        msg = "Subclasses must implement the `fit` method."
+        raise NotImplementedError(msg)
```

Comment:
We might be able to write a generic default implementation for an sklearn model here? Similar to the approach with the [PyTorchBackend](https://github.com/alan-turing-institute/autoemulate/blob/46ca7a6c6b35cdc7cb04e687f89fe8cc921e1f75/autoemulate/experimental/emulators/base.py#L85-L149). For this to work though, I think we'd need to add `.model_` attribute to the backend (or inherit directly from a model class).

Reply:
Yes, I think it will make more sense how to do this once I've added a few other models